core/lib/multivm/src/versions/vm_latest/tests/l1_l2_complete_tx_execution.rs

```diff
@@ -0,0 +1,602 @@
+    );
+
+    bridgehub_contract
+}
+
+#[test]
+fn test_l1_l2_complete_tx_execution_many_small_factory_deps() {
+    // In this test, we try to execute a transaction from L1 to L2 via Gateway
+    // Here instead of marking code hash via the bootloader means, we will be
+    // using L1->Gateway->L2 communication, the same it would likely be done during the priority mode.
+
+    let mut vm = VmTesterBuilder::new(HistoryEnabled)
+        .with_empty_in_memory_storage()
+        .with_base_system_smart_contracts(BASE_SYSTEM_CONTRACTS.clone())
+        .with_execution_mode(TxExecutionMode::VerifyExecute)
+        .with_random_rich_accounts(1)
+        .build();
+
+    let default_account: Account = vm.rich_accounts[0].clone();
+
+    let bridgehub_contract =
+        prepare_environment_and_deploy_contracts(&mut vm, default_account.address);
+
+    // Collect Data for Test
+
+    let request_params = BridgeHubRequestL2TransactionOnGateway::default();
+
+    // Generate a large number of vectors
+    let small_vector_size: usize = 32; // Size of each vector
+    let num_vectors = 10_676; // Number of vectors
+
+    let factory_deps: Vec<Vec<u8>> = (0..num_vectors)
+        .map(|_| vec![0u8; small_vector_size])
+        .collect();
+
+    // Update request_params with the current factory_deps
+    let mut modified_request_params = request_params.clone();
+    modified_request_params.factory_deps = factory_deps.clone();
+
+    let encoded_data = bridgehub_contract
+        .function("forwardTransactionOnGateway")
+        .unwrap()
+        .encode_input(&modified_request_params.to_tokens())
+        .unwrap();
+
+    println!("Factory deps data size: {}", factory_deps.len());
+    println!("Encoded data size: {}", encoded_data.len());
+
+    send_prank_tx_and_verify(
+        &mut vm,
+        L2_BRIDGEHUB_ADDRESS,
+        vec![],
+        encoded_data,
+        SETTLEMENT_LAYER_RELAY_SENDER,
+    );
+}
+
+#[test]
+fn test_l1_l2_complete_tx_execution_few_large_factory_deps() {
+    // In this test, we try to execute a transaction from L1 to L2 via Gateway
+    // Here instead of marking code hash via the bootloader means, we will be
+    // using L1->Gateway->L2 communication, the same it would likely be done during the priority mode.
+
+    let mut vm = VmTesterBuilder::new(HistoryEnabled)
+        .with_empty_in_memory_storage()
+        .with_base_system_smart_contracts(BASE_SYSTEM_CONTRACTS.clone())
+        .with_execution_mode(TxExecutionMode::VerifyExecute)
+        .with_random_rich_accounts(1)
+        .build();
+
+    let default_account: Account = vm.rich_accounts[0].clone();
+
+    let bridgehub_contract =
+        prepare_environment_and_deploy_contracts(&mut vm, default_account.address);
+
+    // Collect Data for Test
+
+    let request_params = BridgeHubRequestL2TransactionOnGateway::default();
+
+    // Generate a large number of vectors
+    let small_vector_size: usize = 45_824; // Size of each vector
```

Comment:
This is the maximal possible size, right?

Reply:
yes